### ISSUE TITLE
ApplicativeAsk with subtyping for ZIO

### ISSUE THREAD

**Issue by danslapman** [zio/interop-cats#231]
Motivation:
`R` in `ZIO[R, .., ..]` frequently holds complex environments (`Has[A] with Has[B] with ...`) while ApplicativeAsk designed to be an atomic context bound (`ApplicativeAsk[F, Has[A]]`). Currently interop-cats provides only ApplicativeLocal instance for the whole invironment, which is inconvenient in many cases.

Proposal:
add an ApplicativeAsk instance with support of subtyping, something like that:

```scala
implicit def zioApplicativeAsk[R1, R <: R1, E](implicit ev: Applicative[ZIO[R, E, ?]]): ApplicativeAsk[ZIO[R, E, ?], R1] =
    new DefaultApplicativeAsk[ZIO[R, E, ?], R1] {
      override val applicative: Applicative[ZIO[R, E, *]] = ev
      override def ask: ZIO[R, E, R1] = ZIO.environment
    }
```

I already started to work on PR with subtyping instance, but it will be ambiguous to existing ApplicatioveLocal instance

**Comment by joroKr21** [comment]
Hmm are you sure it will be ambiguous? I would expect `ApplicativeLocal` to be selected over `ApplicativeAsk` when applicable (because it's a subtype).

On another note, `cats-mtl` 1.0 got released so it might be worth it to first update (there are breaking changes).

**Comment by danslapman** [comment]
@danslapman you're right, ApplicativeLocal will be selected, I just opened #232 

As for cats-mtl 1.0, I will open a separate PR for that

**Comment by danslapman** [comment]
Fixed in #232
